File: 2024/day1P2.pl
#!/usr/bin/perl
use strict;
use warnings;

use Path::Tiny;
use autodie;    # die if problem reading or writing a file
use List::Util qw(sum);

my $dir = path("./");    # /tmp

my $file = $dir->child("day1Input.txt");

# Read in the entire contents of a file
my $content = $file->slurp_utf8();

# openr_utf8() returns an IO::File object to read from
# with a UTF-8 decoding layer
my $file_handle = $file->openr_utf8();

my @left_list  = ();
my @right_list = ();

# Read in line at a time
while ( my $line = $file_handle->getline() ) {
    my @spl = split( /\s+/, $line );
    push @left_list,  $spl[0];
    push @right_list, $spl[1];

}

my @distances_list = ();

foreach my $item (@left_list) {
    my $count = grep { $_ eq $item } @right_list;
    push @distances_list, ( $count * $item );
}

print sum(@distances_list);

File: 2024/day1P2.py
import re

# Define the file's name.
filename = "day1Input.txt"

# Open the file and read its content.
with open(filename) as f:
    content = f.read().splitlines()

listOne = []
listTwo = []
# Display the content.
for index, line in enumerate(content):
    try:
        lineArray = line.split()
        listOne.append(int(lineArray[0]))
        listTwo.append(int(lineArray[1]))
    except IOError:
        pass

distanceArray = []
for index, itemOne in enumerate(listOne):
    try:
        occurrence = listTwo.count(itemOne)
        distanceArray.append(occurrence*itemOne)
    except IOError:
        pass
print(sum(distanceArray))
